packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -151,51 +179,67 @@ export interface BulkAddData {
 }
 
-export interface EditableGridProps {
+export interface SharedEditableGridProps {
+    addControlProps?: Partial<AddRowsControlProps>;
     allowAdd?: boolean;
     allowBulkAdd?: boolean;
     allowBulkRemove?: boolean;
     allowBulkUpdate?: boolean;
     allowFieldDisable?: boolean;
+    allowRemove?: boolean;
     bordered?: boolean;
-    onBulkAdd?: (data: OrderedMap<string, any>) => BulkAddData;
     bulkAddProps?: Partial<QueryInfoFormProps>;
-    bulkUpdateProps?: Partial<QueryInfoFormProps>;
-    condensed?: boolean;
-    addControlProps?: Partial<AddRowsControlProps>;
-    allowRemove?: boolean;
     bulkAddText?: string;
     bulkRemoveText?: string;
+    bulkUpdateProps?: Partial<QueryInfoFormProps>;
     bulkUpdateText?: string;
     columnMetadata?: Map<string, EditableColumnMetadata>;
+    condensed?: boolean;
     disabled?: boolean;
+    emptyGridMsg?: string;
     forUpdate?: boolean;
-    readOnlyColumns?: List<string>;
-    removeColumnTitle?: string;
+    insertColumns?: List<QueryColumn>;
+    hideCountCol?: boolean;
+    isSubmitting?: boolean;
+    maxRows?: number;
     notDeletable?: List<any>; // list of key values that cannot be deleted.
+    processBulkData?: (data: OrderedMap<string, any>) => BulkAddData;
+    readOnlyColumns?: List<string>;
     readonlyRows?: List<any>; // list of key values for rows that are readonly.
-    striped?: boolean;
-    initialEmptyRowCount?: number;
-    model: QueryGridModel;
-    isSubmitting?: boolean;
-    onRowCountChange?: (rowCount?: number) => any;
-    emptyGridMsg?: string;
-    maxTotalRows?: number;
-    hideCountCol?: boolean;
+    removeColumnTitle?: string;
     rowNumColumn?: GridColumn;
-    onCellModify?: () => any;
-    getInsertColumns?: () => List<QueryColumn>;
-    getUpdateColumns?: () => List<QueryColumn>;
+    striped?: boolean;
+    updateColumns?: List<QueryColumn>;
+}
+
+export interface EditableGridProps extends SharedEditableGridProps {
+    data?: Map<any, Map<string, any>>;
+    dataKeys?: List<any>;
+    editorModel: EditorModel;
+    error: string;
+    onChange: (
+        editorModelChanges: Partial<EditorModelProps>,
+        dataKeys?: List<any>,
+        data?: Map<any, Map<string, any>>
+    ) => void;
+    queryInfo: QueryInfo;
 }
 
 export interface EditableGridState {
-    selected: Set<string>;
+    selected: Set<number>;
     selectedState: GRID_CHECKBOX_OPTIONS;
     showBulkAdd: boolean;
     showBulkUpdate: boolean;
+    showMask: boolean;
 }
 
-export class EditableGrid extends ReactN.PureComponent<EditableGridProps, EditableGridState, GlobalAppState> {
+// TODO: move paste events and related code into this component so we can remove them from actions.ts. This can probably
+//  be postponed though, we're far along enough to start work on a QM version of EditableGridPanel, it should only be
+//  done first if we know for sure it will assist with merging with Susan's work which as not been confirmed. We could
```

Comment:
this comment mentions Susan's work / merge. I assume that has already happened. If so, does this comment still apply?

Reply:
Yes old comment!